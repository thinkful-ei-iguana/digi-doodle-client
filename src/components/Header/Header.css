.prompt {
    color: #f55440;
}
.header-box {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    align-content: center;
    margin: 10px auto;
}
h4 {
    margin: 0 30px;
}
.player-prompt {
    margin-top: -2px;
    margin-bottom: 10px;
}
@media screen and (min-width: 280px) {
    h4 {
      font-size: 18px;
    }
    .player-prompt {
        margin-top: 1%;
    }
    .username-display {
        margin-bottom: 3%;
    }
    .header-box {
        margin-bottom: -2%;
    }
  }
  @media screen and (min-width: 400px) {
    .header-box {
        margin-top: 100px;
        margin-bottom: -1%;
    }
  }
  @media screen and (min-width: 550px) {
    .player-prompt {
        margin-top: 2%;
    }
    .header-box {
        margin-top: 2%;
        margin-bottom: -1%;
    }
  }
  @media screen and (min-width: 700px) {
      .header-box {
          margin-bottom: 2%;
      }
  }
  @media screen and (min-width: 800px) {
    .player-prompt {
        margin-top: -2px;
        margin-bottom: 10px;
    }
  }
  @media screen and (min-width: 1000px) {
    h4 {
      font-size: 30px;
    } 
  }

