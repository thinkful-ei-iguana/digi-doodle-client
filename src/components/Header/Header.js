import React, { Component } from 'react'
import ColorContext from '../../Context/ColorContext'
import Timer from '../Timer/Timer';
import './Header.css'

export default class StandbyView extends Component {
    static contextType = ColorContext
    constructor(props) {
        super(props);
        this.state = {
            guess: '',
            username: '',
            players: [],
            score: 0,
            messages: [{
                player: 'Lobby',
                message: 'Welcome to the room!'
            }],
            current_drawer: '',
        }
    }

    static contextType = ColorContext


    render() {
        let header;
        let player = this.context.players.find(player => {
            return player.player_id === this.context.game.current_drawer
        })

        // results mode
        if (this.context.game.status === 'standby' && this.context.roundResults) {
            header = <h1 style={{ color: 'green' }}>{this.context.roundResults}</h1>;
        }

        //winning mode
        else if (this.context.game.status === 'standby' && this.context.game.winner !== null) {
            header =
                <h1>{this.context.game.winner}, You Win<span className="exclamation">!</span></h1>
        }

        //standby before drawer comes up
        else if (this.context.game.status === 'standby' && this.context.game.winner === null && this.context.roundResults === null) {
            header =
                <div className="header-box">
                    <h1>{player.username}, you're up next to draw</h1>
                    <Timer />
                </div>
        }


        // waiting mode
        else if (this.props.isDrawing === false && this.context.game.status === 'waiting for players') {
            header =
                <div className="header-box">
                    <h1>Waiting for more players</h1>
                </div>
        }

        // guessing mode
        else if (this.props.isDrawing === false && this.context.game.status === 'drawing') {

            header = 
            <div className="header-box">
                <h1>{player.username} is drawing</h1>
                <Timer />
            </div>
        }

        // drawing mode
        else if (this.props.isDrawing === true && this.context.game.status === 'drawing') {

            header =
                <div className="header-box">
                    <div className="player-prompt">
                        <h1>{player.username}, Draw <span className="prompt">{this.context.game.current_answer}</span></h1>
                    </div>
                    <Timer />
                </div>
        }

        // waiting mode 2
        else if (this.props.isDrawing === true && this.context.game.status === 'waiting for players') {
            header =
                <div className="header-box">
                    <h1>Draw Something while you wait<span className="exclamation">!</span></h1>
                </div>
        }

        else {
            header = <h1>Something went wrong, please go back to <a className="home-link" label="back to home page" href="/">Home</a> and try again.</h1>
        }


        return (
            <div>
                {header}
            </div>
        )

    }
}
