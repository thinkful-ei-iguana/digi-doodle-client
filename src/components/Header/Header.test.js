import React from 'react'
import ReactDOM from 'react-dom'
import Header from './Header'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

describe(`Header component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>, div);

        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the appropriate header', () => {
        const context = {
            game: {
                id: 'game uuid',
                current_drawer: null,
                current_answer: null,
                status: 'waiting for players',
                winner: null
            }
        }

        const props = {
            isDrawing: false
        }

        const header = renderer.create(<Header {...props} />, context).toJSON();
        expect(header).toMatchSnapshot();
    })

})