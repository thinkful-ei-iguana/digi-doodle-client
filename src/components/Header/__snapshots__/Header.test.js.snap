// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the appropriate header 1`] = `
<div>
  <h1>
    Something went wrong with the game data.
  </h1>
</div>
`;
